packages/@glimmer/compiler/lib/template-compiler.ts

```diff
@@ -443,6 +459,24 @@ function isArg(path: AST.PathExpression): boolean {
   return !!path['data'];
 }
 
+function isDynamicComponent(element: AST.ElementNode): boolean {
+  let open = element.tag.charAt(0);
+
+  let isNamedArgument = open === '@';
+  let isLocal = element['symbols'].has(element.tag);
```

Comment:
Why do we need to use bracket notation here? Does `symbols` need to be added to the type?